Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -471,35 +476,51 @@ extension ChatChannel {
         }()
 
         let messages: [ChatMessage] = {
-            MessageDTO
-                .load(
-                    for: dto.cid,
-                    limit: dto.managedObjectContext?.localCachingSettings?.chatChannel.latestMessagesLimit ?? 25,
-                    deletedMessagesVisibility: dto.managedObjectContext?.deletedMessagesVisibility ?? .visibleForCurrentUser,
-                    shouldShowShadowedMessages: dto.managedObjectContext?.shouldShowShadowedMessages ?? false,
-                    context: context
-                )
+            guard !dto.messages.isEmpty else { return [] }
+            let request = MessageDTO.channelMessagesRequest(
+                for: dto.cid,
+                limit: dto.managedObjectContext?.localCachingSettings?.chatChannel.latestMessagesLimit ?? 25,
+                deletedMessagesVisibility: dto.managedObjectContext?.deletedMessagesVisibility ?? .visibleForCurrentUser,
+                shouldShowShadowedMessages: dto.managedObjectContext?.shouldShowShadowedMessages ?? false
+            )
+            return dto.messages
+                .filtered(using: request)
                 .compactMap { try? $0.relationshipAsModel(depth: depth) }
         }()
 
         let latestMessageFromUser: ChatMessage? = {
             guard let currentUser = context.currentUser else { return nil }
-
-            return try? MessageDTO
-                .loadLastMessage(
-                    from: currentUser.user.id,
-                    in: dto.cid,
-                    context: context
-                )?
+            guard !dto.messages.isEmpty else { return nil }
+            let request = MessageDTO.lastMessageRequest(
+                from: currentUser.user.id,
+                in: dto.cid,
+                context: context
+            )
+            return try? dto.messages
+                .filtered(using: request)
+                .first?
```

Comment:
We probably can simplify this. The backend returns the latestMessages for us. It looks like we are ignoring it and doing unnecessary query logic for this. Ideally, we just need to store the latestMessages in a new `MessageDTO.latestMessages` and then map it directly into `ChatChannel.latestMessages` but sort it by `createdAt`

Reply:
Removed the request part and asking for this information from the `dto.messages` directly.